### DISCUSSION TITLE
Feature Request: auto-sync your forks with upstream

### DISCUSSION THREAD

**Discussion by nutterthanos** [community/community#23001] [New to GitHub]
<p>Hello this is my first post on GitHub Community.</p>
<p>I have a feature request that might interest some people hope like in the 10s of Thousands.</p>
<p>Auto-sync of repo with a option to toggle it in each respective repo.</p>
<p>hope it gets added because it sounds like a good feature to have in case you need the bug fixes from upstream (as long as it does not break copyright)</p>
<p>if this feature does not get added i won’t worry about it then</p>
<p>Thanks</p>
<p>nutterthanos</p>

**Comment by BelHFL** [comment]
<p>Hi <a class="mention" href="/nutterthanos">@nutterthanos</a>, </p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Cheers!</p>

**Comment by nutterthanos** [comment]
<p>Thanks for sharing this with the appropriate teams<br>
Thanks nutterthanos</p>

**Comment by wei** [comment]
<p>My Github App Does exactly what you want : )</p>
<p><a href="https://github.com/wei/pull" rel="nofollow noopener">https://github.com/wei/pull</a></p>
<p>:robot: a GitHub App built with probot that keeps your repository up-to-date with upstream changes via automated PRs.</p>

**Reply by Jay-o-Way** [reply]
So, if I understand correct, that works on "fork-side", right? Because I'm helping at PowerToys and that repo has 4.8k forks and 34 branches - of which at least 10 are stale. But this can't be "installed" in the PowerToys repo itself?

**Reply by wei** [reply]
@Jay-o-Way yes the typically use-case is fork-side. 

However, the advanced configuration supports syncing up branches within your repo as well even if it's not a fork.

**Comment by nutterthanos** [comment]
<p>That works awesome app but just wish it was a official feature not a unofficial feature.</p>

**Comment by muescha** [comment]
<p>i would also see this without creating additional pull requests</p>

**Comment by muescha** [comment]
<p>just note: i installed it and it works like expected. pull request merged automatic. and commit history is clean :slight_smile:</p>
<p>thx for the app :slight_smile:</p>
